L1Trigger/TrackFindingTracklet/interface/DR.h

```diff
@@ -0,0 +1,80 @@
+#ifndef L1Trigger_TrackFindingTracklet_DR_h
+#define L1Trigger_TrackFindingTracklet_DR_h
+
+#include "L1Trigger/TrackTrigger/interface/Setup.h"
+#include "L1Trigger/TrackerTFP/interface/DataFormats.h"
+#include "L1Trigger/TrackFindingTracklet/interface/ChannelAssignment.h"
+
+#include <vector>
+
+namespace trklet {
+
+  /*! \class  trklet::DR
+   *  \brief  Class to emulate Duplicate Removal f/w
+   *  \author Thomas Schuh
+   *  \date   2023, Feb
+   */
+  class DR {
+  public:
+    DR(const edm::ParameterSet& iConfig,
+       const tt::Setup* setup_,
+       const trackerTFP::DataFormats* dataFormats,
+       const ChannelAssignment* channelAssignment,
+       int region);
+    ~DR() {}
+    // read in and organize input tracks and stubs
+    void consume(const tt::StreamsTrack& streamsTrack, const tt::StreamsStub& streamsStub);
+    // fill output products
+    void produce(tt::StreamsStub& accpetedStubs,
+                 tt::StreamsTrack& acceptedTracks,
+                 tt::StreamsStub& lostStubs,
+                 tt::StreamsTrack& lostTracks);
+
+  private:
+    struct Stub {
+      Stub(const tt::FrameStub& frame, int layerId, int stubId, int channel)
+          : frame_(frame), layerId_(layerId), stubId_(stubId), channel_(channel) {}
+      bool operator==(const Stub& s) const {
+        // we ignore fake seed stubs for the moment, to be romved when real seed stubs are available at TrackBuilder output
+        if (s.stubId_ == -1 || stubId_ == -1)
+          return false;
+        return s.layerId_ == layerId_ && s.stubId_ == stubId_;
+      }
+      tt::FrameStub frame_;
+      // det layer id [0-5] barrel [6-10] endcap discs
+      int layerId_;
+      // all stubs id
+      int stubId_;
+      // kf layer id
+      int channel_;
+    };
+    struct Track {
+      static constexpr int max_ = 7;
```

Comment:
Can you add a comment to clarify there are max 7 stubs per layer?

Reply:
there are only one stub per layer but max 7 layers a track may cross, comment added.